v3/newrelic/tracing.go

```diff
@@ -843,17 +847,26 @@ func endDatastoreSegment(p endDatastoreParams) error {
 		evt.Category = spanCategoryDatastore
 		evt.Kind = "client"
 		evt.Component = p.Product
-		evt.AgentAttributes.addString(SpanAttributeDBStatement, p.ParameterizedQuery)
-		evt.AgentAttributes.addString(SpanAttributeDBInstance, p.Database)
-		evt.AgentAttributes.addString(SpanAttributePeerAddress, datastoreSpanAddress(p.Host, p.PortPathOrID))
-		evt.AgentAttributes.addString(SpanAttributePeerHostname, p.Host)
-		evt.AgentAttributes.addString(SpanAttributeDBCollection, p.Collection)
+		evt.AgentAttributes.addString(SpanAttributeDBStatement, truncateSpanAttribute(p.ParameterizedQuery, MaxSpanAttributeDBStatementSize), MaxSpanAttributeDBStatementSize)
+		evt.AgentAttributes.addString(SpanAttributeDBInstance, p.Database, attributeValueLengthLimit)
+		evt.AgentAttributes.addString(SpanAttributePeerAddress, datastoreSpanAddress(p.Host, p.PortPathOrID), attributeValueLengthLimit)
+		evt.AgentAttributes.addString(SpanAttributePeerHostname, p.Host, attributeValueLengthLimit)
+		evt.AgentAttributes.addString(SpanAttributeDBCollection, p.Collection, attributeValueLengthLimit)
 		p.TxnData.saveSpanEvent(evt)
 	}
 
 	return err
 }
 
+func truncateSpanAttribute(value string, maxLength int) string {
+
+	if len(value) > maxLength {
```

Comment:
So if `maxLength` is `4096`, and `len(value)` is something like `4097`, we would still throw away the last 3 bytes of data to replace with `...`, even though we're only one byte of data over the limit? In other words, intentionally toss 2 otherwise valid bytes of data?

Reply:
nvm, found the spec and this is definitely intended